rn/Teacher/src/modules/files/ViewFile.js

```diff
@@ -284,6 +285,26 @@ export default class ViewFile extends Component<Props, State> {
             </Text>
           </View>
         )
+      case 'html':
+        // some people use files as a web server to serve html content and
+        // when they do they might include relative links to other files in the course
+        // which won't work with the download url. We must build a preview url that will
+        // redirect the webview to the server that serves file content in order for the
+        // relative urls to function properly
+        let { baseURL } = getSession()
+        let url = baseURL
+        if (this.props.context && this.props.contextID) {
+          url += `${this.props.context}/${this.props.contextID}/`
```

Comment:
I believe file ids are unique for the account. I don't think we need to append the context. Do you know for sure?

Reply:
Yeah when I tried to do the preview in the browser with just `file/:fileID/preview` it failed.
https://ckruger.instructure.com/files/48901/preview
https://ckruger.instructure.com/courses/3313/files/48901/preview

Reply:
I do believe they are unique but the rails route/controller is probably set up to expect a context if it belongs to one or something like that.

Reply:
Good to know! Thanks for clarifying.